slack-message.c

```diff
@@ -479,7 +479,7 @@ void slack_handle_message(SlackAccount *sa, SlackObject *obj, json_value *json,
 
 static void handle_message(SlackAccount *sa, gpointer data, SlackObject *obj) {
 	json_value *json = data;
-	return slack_handle_message(sa, obj, json, PURPLE_MESSAGE_RECV);
+	slack_handle_message(sa, obj, json, PURPLE_MESSAGE_RECV);
```

Comment:
This does seem suspicious, but despite the fact that I feel like I only would have done this with good reason, it does seem that you're right and this needs freeing... Oops!